dspace-api/src/main/java/org/dspace/importer/external/service/components/QuerySource.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.importer.external.service.components;
+
+import java.util.Collection;
+
+import org.dspace.content.Item;
+import org.dspace.importer.external.datamodel.ImportRecord;
+import org.dspace.importer.external.datamodel.Query;
+import org.dspace.importer.external.exception.MetadataSourceException;
+
+
+/**
+ * Common interface for database-based imports.
+ *
+ * @author Roeland Dillen (roeland at atmire dot com)
+ * @author Pasquale Cavallo (pasquale.cavallo@4science.it)
+ */
+
+public interface QuerySource extends MetadataSource {
+
+    /**
+     * Get a single record from the source.
+     * The first match will be returned
+     *
+     * @param id identifier for the record
+     * @return a matching record
+     * @throws MetadataSourceException if the underlying methods throw any exception.
+     */
+    public ImportRecord getRecord(String id) throws MetadataSourceException;
+
+    /**
+     * Gets the number of records matching a query
+     *
+     * @param query the query in string format
+     * @return the number of records matching the query
+     * @throws MetadataSourceException if the underlying methods throw any exception.
+     */
+    public int getNbRecords(String query) throws MetadataSourceException;
```

Comment:
Tiny thing.  We may want to call this `getRecordsCount`.... I initially didn't understand what "Nb" stands for, but it looks like it's the number/count of records.